calendar/engine/availability.go

```diff
@@ -279,37 +280,124 @@ func (m *mscalendar) setUserStatuses(users []*store.User, calendarViews []*remot
 		}
 
 		var err error
-		res, isStatusChanged, err = m.setStatusFromCalendarView(user, status, view)
-		if err != nil {
-			if numberOfLogs < logTruncateLimit {
-				m.Logger.Warnf("Error setting user %s status. err=%v", user.MattermostUserID, err)
-			} else if numberOfLogs == logTruncateLimit {
-				m.Logger.Warnf(logTruncateMsg)
+		if user.Settings.UpdateStatusFromOptions != NotSetStatusOption {
+			res, isStatusChanged, err = m.setStatusFromCalendarView(user, status, view)
+			if err != nil {
+				if numberOfLogs < logTruncateLimit {
+					m.Logger.Warnf("Error setting user %s status. err=%v", user.MattermostUserID, err)
+				} else if numberOfLogs == logTruncateLimit {
+					m.Logger.Warnf(logTruncateMsg)
+				}
+				numberOfLogs++
+				numberOfUserErrorInStatusChange++
+			}
+			if isStatusChanged {
+				numberOfUserStatusChange++
 			}
-			numberOfLogs++
-			numberOfUserErrorInStatusChange++
 		}
-		if isStatusChanged {
-			numberOfUserStatusChange++
+
+		if user.Settings.SetCustomStatus {
+			res, isStatusChanged, err = m.setCustomStatusFromCalendarView(user, view)
+			if err != nil {
+				if numberOfLogs < logTruncateLimit {
+					m.Logger.Warnf("Error setting user %s status. err=%v", user.MattermostUserID, err)
+				} else if numberOfLogs == logTruncateLimit {
+					m.Logger.Warnf(logTruncateMsg)
+				}
+				numberOfLogs++
+				numberOfUserErrorInStatusChange++
+			}
+
+			// Increment count only when we have not updated the status of the user from the options to have status change count per user.
+			if isStatusChanged && user.Settings.UpdateStatusFromOptions == NotSetStatusOption {
+				numberOfUserStatusChange++
+			}
 		}
 	}
+
 	if res != "" {
 		return res, numberOfUserStatusChange, numberOfUserErrorInStatusChange, nil
 	}
 
 	return utils.JSONBlock(calendarViews), numberOfUserStatusChange, numberOfUserErrorInStatusChange, nil
 }
 
+func (m *mscalendar) setCustomStatusFromCalendarView(user *store.User, res *remote.ViewCalendarResponse) (string, bool, error) {
+	isStatusChanged := false
+	if !user.Settings.SetCustomStatus {
+		return "User doesn't want to set custom status", isStatusChanged, nil
+	}
+
+	events := filterBusyEvents(res.Events)
+	if len(events) == 0 {
+		if user.IsCustomStatusSet {
+			if err := m.PluginAPI.RemoveMattermostUserCustomStatus(user.MattermostUserID); err != nil {
+				m.Logger.Warnf("Error removing user %s custom status. err=%v", user.MattermostUserID, err)
+			}
+
+			if err := m.Store.StoreUserCustomStatusUpdates(user.MattermostUserID, false); err != nil {
+				return "", isStatusChanged, err
+			}
+		}
+
+		return "No event present to set custom status", isStatusChanged, nil
+	}
+
+	if checkOverlappingEvents(events) {
+		return "Overlapping events, not setting a custom status", isStatusChanged, nil
+	}
```

Comment:
I continued the conversation for this here https://github.com/mattermost/mattermost-plugin-mscalendar/issues/228#issuecomment-2123280460

Reply:
@mickmister This was done because we were ignoring meetings as they were overlapping. But the approach will be changed now. 
What should we do when we have two meetings that are not overlapping? For eg: A meeting timings: 1:00-1:02, B meeting timings: 1:03-1:05. As we will be receiving both of them at the same time, should we create a job to update user status for meeting B?

Reply:
@ayusht2810 I trust your judgment on how to go about handling overlaps etc. I have to say I don't fully understand

Reply:
If we have overlapping meetings, can we use the times from all meetings combined? From your example: 1.00 to 1.05.

Reply:
@fmartingr Yes, I was also thinking of the same approach. I will update the PR with this change.